core/src/main/java/com/linecorp/armeria/internal/client/DefaultClientRequestContext.java

```diff
@@ -438,7 +440,7 @@ private void failEarly(Throwable cause) {
     }
 
     // TODO(ikhoon): Consider moving the logic for filling authority to `HttpClientDelegate.exceute()`.
-    private void autoFillSchemeAndAuthority() {
+    private void autoFillSchemeAuthorityAndOrigin() {
```

Comment:
While double checking, it seems like the rfc specifies origin as optional.

> 7.   Optionally, an |Origin| header field.  This header field is sent
        by all browser clients.  A connection attempt lacking this
        header field SHOULD NOT be interpreted as coming from a browser
        client.

I'm fine with this change if this is important for compatibility with other server implementations, but I wanted to double-check if we want to add this header by default.

ref: https://datatracker.ietf.org/doc/html/rfc6455#section-4.2.1